test/routing/query-params.test.js

```diff
@@ -1,72 +1,128 @@
 var test  = require('tape')
   , rhumb = require('../../src/rhumb')
+  , utils = require('../utils')
 
 test("Routing should pass query string params to callback", function(t) {
-  var router = rhumb.create()
+  t.plan(2)
 
-  t.plan(1)
-  router.add("/sing", function(params) {
-    t.deepEquals(params,
-      { foo: "bar"
-      , baz: "bam"
-      }
-    )
+  utils.performMatch('/foo', '/foo?q=value', function (params) {
+    t.deepEquals(params, { q: 'value' }
+      , 'path should match with query params included in params list')
   })
 
-  router.match("/sing?foo=bar&baz=bam")
+  utils.performMatch('/foo/bar/bing', '/foo/bar/bing?q=value', function (params) {
+    t.deepEquals(params, { q: 'value' }
+      , 'path should match with query params included in params list')
+  })
 })
 
 test("Routing should pass query string params and path params to callback", function(t) {
-  var router = rhumb.create()
-
   t.plan(1)
-  router.add("/sing/{sound}", function(params) {
-    t.deepEquals(params,
-      { sound : "bird-song"
-      , foo : "bar"
-      , baz : "bam"
-      }
-    )
-  })
 
-  router.match("/sing/bird-song?foo=bar&baz=bam")
+  utils.performMatch('/wibble/{wobble}', '/wibble/foo?q=value', function (params) {
+    t.deepEqual(params, { wobble: 'foo', q: 'value' }
+      , 'path should match without overwriting path params with query params')
```

Comment:
It seems to me that the test description isn't what we're testing here. Rather, we should probably test something like `/wibble/{wobble}` == `/wibble/foo?wobble=value` and assert that `wobble === 'foo'`.

This would mean though that query params are effectively shadowed by segment variables. This is nothing new, and I believe #5 already describes the problem in more detail.